src/hooks/use-form.js

```diff
@@ -99,13 +99,13 @@ const valuesReducer = (state, action) => {
     case actionTypes.SET_FIELD_VALUE:
       return {
         ...state,
-        [payload.field]: payload.value
+        ...set(state, payload.field, payload.value)
```

Comment:
`set` mutates the object you pass as the first argument, so this mutates the previous state, which should be avoided. Instead, please pass an empty object as the first argument:

```suggestion
        ...set({}, payload.field, payload.value)
```

Please do the same on other `set` calls you have.